src/beamlime/dashboard/widgets/parameter_widget.py

```diff
@@ -0,0 +1,74 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2025 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+from typing import Any
+
+import panel as pn
+
+from beamlime.config.workflow_spec import Parameter, ParameterType
+
+
+class ParameterWidget:
+    """Widget for configuring a single workflow parameter."""
+
+    def __init__(self, parameter: Parameter, initial_value: Any | None = None) -> None:
+        """
+        Initialize parameter widget.
+
+        Parameters
+        ----------
+        parameter
+            Parameter specification
+        initial_value
+            Initial value for the parameter, if None uses parameter default
+        """
+        self._parameter = parameter
+        self._value = initial_value if initial_value is not None else parameter.default
+        self._widget = self._create_widget()
+
+    def _create_widget(self) -> pn.Param:
+        """Create the appropriate Panel widget based on parameter type."""
+        widget_config = {
+            "name": self._parameter.name,
+            "value": self._value,
+        }
+
+        if self._parameter.unit:
+            widget_config["name"] = f"{self._parameter.name} ({self._parameter.unit})"
```

Comment:
```suggestion
            widget_config["name"] = f"{self._parameter.name} [{self._parameter.unit}]"
```

Reply:
But isn't having two styles in the same GUI confusing? Or are you suggesting we should try to override how Holoviews shows units?

Reply:
> Or are you suggesting we should try to override how Holoviews shows units?

Yes. If this is simple enough. Otherwise, go with parentheses.

Reply:
See #399.